### PR TITLE
Rename HTTPTextFormat to TextMapPropagator

### LINKED ISSUE #536: Rename HTTPTextFormat to TextMapPropagator
As per the specs:
https://github.com/open-telemetry/opentelemetry-specification/blob/b17f8c993af285cfb0b9cfeb872c15911c720ad2/specification/context/api-propagators.md#textmap-propagator

The relevant specs PR : https://github.com/open-telemetry/opentelemetry-specification/pull/793

### PR DESCRIPTION
**Pull request by TomRoSystems**
Fixes #536

## Changes

As per spec open-telemetry/opentelemetry-specification#793

Trivial change.

### CODE REVIEW

**Review by ThomsonTan — APPROVED** [review]

File: api/include/opentelemetry/trace/propagation/text_map_propagator.h [hunk 1]
```diff
@@ -11,13 +11,13 @@ namespace trace
 namespace propagation
 {
 
-// The HTTPTextFormat class provides an interface that enables extracting and injecting
+// The TextMapPropagator class provides an interface that enables extracting and injecting
 // context into headers of HTTP requests. HTTP frameworks and clients
```
Comment (lalitb) [inline api/include/opentelemetry/trace/propagation/text_map_propagator.h:15]:
```suggestion
// context into carriers that travel in-band across process boundaries. HTTP frameworks and clients
```

File: api/include/opentelemetry/trace/propagation/text_map_propagator.h [hunk 2]
```diff
@@ -11,13 +11,13 @@ namespace trace
...
-// can integrate with HTTPTextFormat by providing the object containing the
+// can integrate with TextMapPropagator by providing the object containing the
 // headers, and a getter and setter function for the extraction and
 // injection of values, respectively.
 template <typename T>
-class HTTPTextFormat
+class TextMapPropagator
```
Comment (lalitb) [inline api/include/opentelemetry/trace/propagation/text_map_propagator.h:20]: Nit: We can change `HTTP header carrier` to `carrier` too in `Extract` and `Inject` functions.

**Review by lalitb — APPROVED** [review]

File: api/include/opentelemetry/trace/propagation/text_map_propagator.h (hunk 2 above)
Comment (TomRoSystems) [reply api/include/opentelemetry/trace/propagation/text_map_propagator.h:20]: Yes - changed as suggested.

File: api/include/opentelemetry/trace/propagation/text_map_propagator.h (hunk 1 above)
Comment (TomRoSystems) [reply api/include/opentelemetry/trace/propagation/text_map_propagator.h]: Yes - changed as suggested.

**Comment by lalitb** [comment]
Thanks for the fix.

**Pull request merged**